server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -249,15 +249,15 @@ public void addMapping(String user, Domain domain, String mapping) throws Recipi
 
         String map = mapping.toLowerCase(Locale.US);
 
-        if (map.startsWith(RecipientRewriteTable.ERROR_PREFIX)) {
-            addErrorMapping(user, domain, map.substring(RecipientRewriteTable.ERROR_PREFIX.length()));
-        } else if (map.startsWith(RecipientRewriteTable.REGEX_PREFIX)) {
-            addRegexMapping(user, domain, map.substring(RecipientRewriteTable.REGEX_PREFIX.length()));
-        } else if (map.startsWith(RecipientRewriteTable.ALIASDOMAIN_PREFIX)) {
+        if (map.startsWith(Type.Error.asPrefix())) {
+            addErrorMapping(user, domain, map.substring(Type.Error.asPrefix().length()));
```

Comment:
substring could be done by Type instance

Reply:
https://github.com/linagora/james-project/pull/1350 did it.